crates/biome_module_graph/src/js_module_info/visitor.rs

```diff
@@ -18,6 +18,14 @@ use crate::{
 
 use super::{ResolvedPath, collector::JsModuleInfoCollector};
 
+/// Extensions to try to resolve based on the extension in the import specifier.
+/// ref: https://www.typescriptlang.org/docs/handbook/modules/reference.html#the-moduleresolution-compiler-option
+const EXTENSION_ALIAS: &[(&str, &[&str])] = &[
+    ("js", &["ts", "tsx", "d.ts", "js", "jsx"]),
+    ("mjs", &["mts", "d.mts", "mjs"]),
+    ("cjs", &["cts", "d.cts", "cjs"]),
```

Comment:
I don't think we should include the extensions themselves as their aliases, since we could already find the correct extensions with the existing logic.

```suggestion
    ("js", &["ts", "tsx", "d.ts", "jsx"]),
    ("mjs", &["mts", "d.mts"]),
    ("cjs", &["cts", "d.cts"]),
```